src/v2/view-builder/internals/FormInputFactory.js

```diff
@@ -33,10 +33,23 @@ const createAuthenticatorEnrollSelectView = (opt) => {
 };
 
 const createAuthenticatorVerifySelectView = (opt) => {
-  var optionItems = (opt.options || [])
-    .map(opt => {
-      return Object.assign({}, opt, FactorUtil.getFactorData(opt.authenticatorType));
-    });
+  let optionItems = (opt.options || []);
+  // If webauthn enrollments > 1 just show one entry with a generic namne (first) so user doesnt have to select which
+  // one to pick. eg) If there is yubikey5 and another unknown u2f key, user cannot identify that easily. We need to
+  // do this at least  until users can give authenticator enrollments custom names.
+  let insertIndex;
+  const webauthnEnrollment = optionItems.find((opt, index) => {
+    if (opt.authenticatorType === 'security_key') {
+      insertIndex = index;
+      return true;
+    }
+    return false;
+  });
+  optionItems = optionItems.filter(opt => opt.authenticatorType !== 'security_key');
+  webauthnEnrollment && optionItems.splice(insertIndex, 0, webauthnEnrollment);
```

Comment:
Is your intent to pick first webauthn in the list? looks like it can be done in one `filter`
```
let hasSecurityKey = false;
optionItems.filter(opt => {

  if opt.authenticatorType === 'security_key' {
     if !hasSecurityKey {
       hasSecurityKey = true;
       return true;
     } else {
       return false;
     }
  }

  return true;
});
```

Reply:
yeah I can use this logic too. Guess it make it simpler.